updated 3rd party packages versions

frankl-uni opened the pull request:
I updated all packages to their actual version, as project does not build with python 3.12

Commit:
updated dependency packages version

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- Updated poetry.lock to match changes in pyproject.toml
- Updated rapidhrv version to 0.2.7

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies.
- Updated poetry.lock to match changes in pyproject.toml
- Updated rapidhrv version to 0.2.7
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies and commented out mypy_plugin (deprecated).
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div and changed app.run_server to app.run.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies and commented out mypy_plugin (deprecated).
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div and changed app.run_server to app.run.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies and commented out mypy_plugin (deprecated).
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div and changed app.run_server to app.run.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies and commented out mypy_plugin (deprecated).
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div and changed app.run_server to app.run.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies and commented out mypy_plugin (deprecated).
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div,changed app.run_server to app.run, and formatting.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies and commented out mypy_plugin (deprecated).
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div,changed app.run_server to app.run, and formatting.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies and commented out mypy_plugin (deprecated).
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div,changed app.run_server to app.run, and formatting.
- preprocessing.py formatting.

peterakirk commented:
Thanks for the updates @frankl-uni! @alxdb, I've made further edits. Everything runs fine and mypy is happy on my local install. However, I can't seem to get Github worklow mypy to be happy with visualization.py when I push. Specifically, it's not happy with the following. LMK if you've any thoughts.

`options: List[OptionType] = [{"label": col, "value": col} for col in rhv.analysis.DATA_COLUMNS]`

alxdb commented:
I think maybe the new type needs to be Options not OptionType?

Just got that from looking at the error output on GitHub mobile.

If that doesn't work, and everything is working as expected, we can just disable type checking for that line with a comment (mypy docs should specify the necessary comment). We can look into the correct type later.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies,  commented out mypy_plugin (deprecated), and instructed mypy to ignore line on visualization.py.
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div,changed app.run_server to app.run, and formatting.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies,  commented out mypy_plugin (deprecated), and instructed mypy to ignore line on visualization.py.
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div,changed app.run_server to app.run, and formatting.
- preprocessing.py formatting.

peterakirk committed:
Updated:
- action.yaml python version.
- pyproject.toml updated poetry.group.dev.dependencies,  commented out mypy_plugin (deprecated), and instructed mypy to ignore line on visualization.py.
- poetry.lock to match changes in pyproject.toml
- rapidhrv version to 0.2.7
- analysis.py trapezoid function
- visualization.py declared options before html.Div,changed app.run_server to app.run, and formatting.
- preprocessing.py formatting.
